form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt

```diff
@@ -532,6 +542,52 @@ private fun ProvideInteger(
     )
 }
 
+@Composable
+private fun ProvideInputPhoneNumber(
+    fieldUiModel: FieldUiModel,
+    intentHandler: (FormIntent) -> Unit,
+    uiEventHandler: (RecyclerViewUiEvents) -> Unit,
+
+) {
+    var value by remember {
+        mutableStateOf(fieldUiModel.value)
+    }
+
+    InputPhoneNumber(
+        modifier = Modifier.fillMaxWidth(),
+        title = fieldUiModel.label,
+        state = when {
+            fieldUiModel.error != null -> InputShellState.ERROR
+            !fieldUiModel.editable -> InputShellState.DISABLED
+            fieldUiModel.focused -> InputShellState.FOCUSED
+            else -> InputShellState.UNFOCUSED
+        },
+        legendData = fieldUiModel.legend?.let {
+            LegendData(Color(it.color), it.label ?: "", null)
+        },
+        inputText = value ?: "",
+        onValueChanged = {
+            value = it
+            intentHandler(
+                FormIntent.OnTextChange(
+                    fieldUiModel.uid,
+                    value,
+                    fieldUiModel.valueType,
+                ),
+            )
+        },
+        onCallActionClicked = {
+            uiEventHandler.invoke(
+                RecyclerViewUiEvents.OpenChooserIntent(
+                    Intent.ACTION_DIAL,
+                    value,
+                    fieldUiModel.uid,
+                ),
+            )
+        },
+    )
```

Comment:
I think that you also need to add supportinText and isRequired properties

Reply:
and just realized that I need another ticket for the phone number as it doesn't allow supporting text just error message...